ngraph/frontend/paddlepaddle/src/op/transpose2.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <ngraph/opsets/opset6.hpp>
+#include <node_context.hpp>
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        namespace pdpd
+        {
+            namespace op
+            {
+                NamedOutputs transpose2(const NodeContext& node)
+                {
+                    auto data = node.get_ng_input("X");
+                    auto perm = node.get_attribute<std::vector<int>>("axis");
+
+                    PDPD_OP_VALIDATION_CHECK(
+                        node,
+                        data.get_partial_shape().rank().is_static(),
+                        "[Frontend]Paddle transpose2 input data rank must be static");
+
+                    auto rank =
+                        static_cast<unsigned long>(data.get_partial_shape().rank().get_length());
+
+                    auto input_order =
+                        ngraph::opset6::Constant::create(ngraph::element::i64, {rank}, perm);
+                    return node.default_single_output_mapping(
+                        {std::make_shared<ngraph::opset6::Transpose>(data, input_order)}, {"Out"});
```

Comment:
```suggestion
                    auto data = node.get_ng_input("X");
                    auto perm = node.get_attribute<std::vector<int>>("axis");
                    auto input_order =
                        ngraph::opset6::Constant::create(ngraph::element::i64, {perm.size()}, perm);
                    return node.default_single_output_mapping(
                        {std::make_shared<ngraph::opset6::Transpose>(data, input_order)}, {"Out"});
```

Reply:
All right, but do we still need to check whether rank is static ?

Reply:
No we don't

Reply:
I suggested to delete it

Reply:
OK, will delete this useless check